Remove unneeded height from account

lucafavatella opened the pull request:
Delivers https://www.pivotaltracker.com/story/show/157308030

Please also review specification doc update at https://github.com/aeternity/protocol/pull/80

----

This PR includes release note for PR #1131 (cc @happi ) in commit:
* Mention contracts state tree changes from PR 1131 in release notes

This PR includes some refactoring / clarifications preparatory for https://www.pivotaltracker.com/story/show/157214052 (move coinbase to header): changes are in separate commits and small though if they make review difficult please let me know and I will remove most of those commits.

Overview of commits:

* Minor genesis-related refactoring. Commits:
  * Refactor height in genesis module
  * Clarify contribution of genesis to chain difficulty

* Minor block state -related refactoring. Commits:
  * Remove unneeded external function call in state pre-transformation
  * Move chain insertion comment back close to the code it refers to
  * Clarify height type and value around state pre-transformations

* A typo - commit "Fix typo in test comment"

* Test improvement preparatory to removal of coinbase transaction. Commits:
  * Move aec_chain unit test re tx in multiple forks away from coinbase tx

* Account height -related changes (**focus of this PR**). Commits:
  * Correct account height-related type
  * **Remove unneeded height from account - MAIN COMMIT**
  * Mention removal of height from account in release notes
  * Use fresh DB on environments for removal of height from account

----

TODO:
* [x] Ensure CI passes.
* [x] Rebase on master once PR #1132 is merged to master.
* [x] Rebase on master once PRs #1131 is merged to master.
  * This seems unavoidable as I expect #1131 to be merged with squashing.
* [x] Remove commits re enabling Python UAT tests in this PR.
  * You can check run of Python UAT tests on CI in [this CI workflow on 847480a0 that is equivalent to ef24ccc8](https://circleci.com/workflow-run/32ef6af5-0e6e-48cb-9d87-81ec1cb9e456) - notice that [Python UAT tests passed there](https://circleci.com/gh/lucafavatella/epoch/3662).
* [ ] ~Remove commits re enabling aevm-tests in this PR~ Master now includes running aevm-tests in PR so no explicit item needed.
* [x] Bump DB folder in environments - may need rebase for conflicts.
* [x] Release note line.

sennui commented on apps/aehttp/test/aehttp_integration_SUITE.erl:
```diff
@@ -4213,7 +4195,7 @@ block_to_endpoint_map(Block, Options) ->
             {BlockHeight, TxBlockHeight} = {TxBlockHeight, BlockHeight},
             {BlockHash, TxBlockHash} = {TxBlockHash, BlockHash},
-            TxHash = aetx:hash(aetx_sign:tx(SignedTx)),
+            TxHash = aetx_sign:hash(SignedTx),
```
why did you decided to move it out from aetx?

lucafavatella replied:
[This comes from PR 1132](https://github.com/aeternity/epoch/pull/1132/files?utf8=%E2%9C%93&diff=unified#diff-14dc0eab04e011248c7640cb3c1cf0deR4198) this PR depends upon (and that is included in this PR - I updated description clarifying inclusion).

lucafavatella commented:
@sennui @zp-sd For ease of review - either you check commit by commit following overview in PR description, or you check https://github.com/aeternity/epoch/compare/1a759efd...788afb47 (that will of course be invalid once I rebase on master later today).

lucafavatella commented:
I force-pushed update:
* Rebased on recent contracts-accounts PR #1131 (actually #1148 that fixes Python UAT tests).
* Fixed arity in a dozen or so invocations to aec_accounts:spend.
* Removed a handful of aec_accounts:height checks in unit tests.
* Added release notes line (will probably require rebase for conflicts).
* Bumped DB folder (may require rebase for conflicts - I bumped by 2 for attempting to prevent conflict for missing bump in current PR 1131).
* Removed commit enabling Python UAT tests in PR.

(For easing comparison, initial version of this PR was 788afb470dc136953f3908045c3e27317fbef1b3 )

----

It looks like there are conflicts for new PRs merged to master - I will rebase again and still include #1148 and force-push again.

lucafavatella commented:
> It looks like there are conflicts for new PRs merged to master - I will rebase again and still include #1148 and force-push again.

Done.

lucafavatella commented:
I am keeping WIP label and assigned to me as waiting for CI runs (including [run with UAT tests](https://circleci.com/workflow-run/32ef6af5-0e6e-48cb-9d87-81ec1cb9e456)) to complete.

lucafavatella commented:
Now ready for review though please expect at least one rebase tomorrow.

lucafavatella commented:
Rebased on recent-master-with-recent-PR-1131-in-it and added, as first commit, release note for PR 1131. Will need to rebase again once 1131 merged.

lucafavatella commented:
Rebased on master now that #1131 merged.
